.github/workflows/main.yaml

```diff
@@ -31,22 +36,51 @@ env:
   SCCACHE_IDLE_TIMEOUT: 0 # prevent sccache server from shutting down after long idle.
   STARBOARD_TOOLCHAINS_DIR: /root/starboard-toolchains
 
+concurrency:
+  group: '${{ github.workflow }}/${{ inputs.platform }} @ ${{ github.event.pull_request.head.label || github.head_ref || github.ref }}'
+  cancel-in-progress: true
+
 # A workflow run is made up of one or more jobs that can run sequentially or in parallel
 jobs:
   # Retrieves configuration from json file.
   initialize:
     runs-on: ubuntu-latest
+    # All triggers except draft PRs, unless PR is labeled with runtest
```

Comment:
What if user wants to run all builds/tests + on device test for their draft PR, will this condition handle it?

Reply:
Yes, as long as the PR has `on_device` label